test/navigation.spec.js

```diff
@@ -352,14 +352,13 @@ module.exports.describe = function({testRunner, expect, playwright, FFOX, CHROMI
       expect(request2.headers['referer']).toBe(undefined);
       expect(page.url()).toBe(server.PREFIX + '/grid.html');
     });
-    it.skip(FFOX)('should fail when canceled by another navigation', async({page, server}) => {
-      server.setRoute('/one-style.css', (req, res) => {});
-      // For some reason, Firefox issues load event with one outstanding request.
-      const failed = page.goto(server.PREFIX + '/one-style.html', { waitUntil: FFOX ? 'networkidle0' : 'load' }).catch(e => e);
-      await server.waitForRequest('/one-style.css');
+    it('should fail when canceled by another navigation', async({page, server}) => {
```

Comment:
Previously, we would reject any navigation promises if the page navigated somewhere before the lifecycle events occurred. But that sometimes had false positives. Consider

```html
<script>
window.location.href = 'https://www.microsoft.com';
</script>
```
This page will redirect to microsoft.com before the 'load' event. But everything is working as intended. This patch changes the behavior to no longer reject the goto promise.

The case this test is testing for can now only be detected if the navigation is interrupted before the document commits. This has a different error message per browser, all variations on the document request being aborted.